lnst/Recipes/ENRT/RDMARecipe.py

```diff
@@ -0,0 +1,79 @@
+from socket import AF_INET
+
+from lnst.Common.IpAddress import interface_addresses
+from lnst.Common.Parameters import IPv4NetworkParam, IntParam, StrParam
+from lnst.Controller import HostReq, DeviceReq, RecipeParam
+from lnst.RecipeCommon.Perf.Measurements import RDMABandwidthMeasurement
+from lnst.RecipeCommon.Perf.Measurements.RDMABandwidthMeasurement import RDMAFlow
+from lnst.RecipeCommon.Perf.Recipe import RecipeConf, Recipe as PerfRecipe
+
+
+class RDMARecipe(PerfRecipe):
+    """
+    RDMA / InfiniBand bandwidth test recipe
+    """
+
+    driver = StrParam(default="mlx5_core")
+    perf_iterations = IntParam(default=5)
+
+    net_ipv4 = IPv4NetworkParam(default="192.168.101.0/24")
+
+    host1 = HostReq()
+    host1.eth0 = DeviceReq(label="net1", driver=RecipeParam("driver"))
+
+    host2 = HostReq()
+    host2.eth0 = DeviceReq(label="net1", driver=RecipeParam("driver"))
+
+    def test(self) -> None:
+        config = self.test_wide_configuration()
+        try:
+            results = self.perf_test(config)
+            self.perf_report(results)
+            # self.perf_evaluate(results)
+        finally:
+            self.test_wide_deconfiguration(config)
+
+    def test_wide_configuration(self) -> RecipeConf:
+        host1, host2 = self.matched.host1, self.matched.host2
+
+        ipv4_addr_pool = interface_addresses(self.params.net_ipv4)
+        for host in [host1, host2]:
+            host.eth0.ip_add(next(ipv4_addr_pool))
+            host.eth0.up()
+
+            # setup RoCE (RDMA over Converged Ethernet), emulating InfiniBand on Ethernet
+            host.run("ip a")
+            host.run("rdma link show")
+            host.run(f"rdma link add ib0 type rxe netdev {host.eth0.name}")
+            host.run("rdma link show")
+
+        flow = RDMAFlow(
+            generator=host1,
+            receiver=host2,
+            generator_nic_name="ib0",
+            receiver_nic_name="ib0",
+            generator_bind=host1.eth0.ips_filter(family=AF_INET)[0],
+            receiver_bind=host2.eth0.ips_filter(family=AF_INET)[0],
+        )
+
+        return RecipeConf(
+            measurements=[
+                RDMABandwidthMeasurement(flows=[flow]),
+            ],
+            iterations=self.params.perf_iterations,
+        )
+
+    def test_wide_deconfiguration(self, config: RecipeConf) -> None:
+        host1, host2 = self.matched.host1, self.matched.host2
+
+        for host in [host1, host2]:
+            host.run(f"rdma link delete ib0")
+
+    def apply_perf_test_tweak(self, config: RecipeConf) -> None:
+        pass
```

Comment:
Does these tweak bypasses need to be specified?

Reply:
Not sure, but now I'm inheriting from `SimpleNetworkRecipe`. This is not relevant anymore.